### ISSUE TITLE
Search params in URL

### ISSUE THREAD

**Issue by graft** [mountetna/monoetna#387]
On the Timur Search page, it would be nice if search parameters showed up in the URL, so that the search could be copied and pasted, allowing it to be saved/shared. Since there are a large number of parameters with complex configurations, how to do it is a question. Two options:

1. Share a base64-encoded JSON. This is simple to implement but does not easily allow editing of params in the URL
 
2. Serialize/deserialize each search param into a url_param. This probably requires a lot of struggle dealing with url encoding issues, but would expose params in some kind of text allowing them to be edited-ish.

I favor (1) since I'm not sure what the actual value is of editing params in the URL. However, it is nice to be able to see what the search is about at a glance. So:

3. Maybe a mix of the two? Leave strings as url_params but base64-encode, say, the filters JSON.

**Comment by coleshaw** [comment]
I agree it would be nice to have visibility into the search params in the URL query params, but I think we should discourage hand-editing of those values -- it's too easy to mess them up and get an invalid URL. If someone changes the model, that invalidates any filters or attribute selections, and there doesn't seem to be an easy way to validate that, if it's just in a URL. I'm not sure there's much value in 3, in that case...

If someone is going to bookmark a URL, I would say let them edit their own bookmark name locally, but go with option 1 above.
